### PR TITLE
PR Milestone 2: Components and Redux store

### PR DESCRIPTION
**Pull request by pastorp3**
## Book Store Milestone 2: Components and Redux store

In this PR I:
- Crete redux store
- Set reducer
- Create actions
- Add components to display the current state
- Integrate redux with components

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,25 +1,12 @@
-import logo from './logo.svg';
 import './App.css';
```
Comment (christianotieno) [inline src/App.js]: ## Move the app component
- [x] The `App.js` component needs to be moved to the `src/components` directory.

File: src/App.js [hunk 2]
```diff
@@ -1,25 +1,12 @@
...
+import BookList from './components/booksList/booksList';
+import BookForm from './components/booksForm/booksForm';
 
 function App() {
```
Comment (christianotieno) [inline src/App.js]: ## Employ the use of ES6 syntax
- [x] Do you mind implementing this functional declaration using the `ES6` arrow syntax?

File: src/components/book/book.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import PropTypes from 'prop-types';
+
+const Book = ({ title, categorie }) => (
+  <tr>
+    <th>{title}</th>
+    <th>{categorie}</th>
+  </tr>
+);
```
Comment (christianotieno) [inline src/components/book/book.js]: ## Implement the missing book id section 
- [x] Our `book` component should represent a single row of books with `id`, `title` and `category` [screenshot from Pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone2/Book/RepresentARowInBooksTable.png). We are currently missing the `id` part, lets add it.

File: src/components/booksForm/booksForm.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
+/* eslint-disable arrow-body-style */
```
Comment (christianotieno) [inline src/components/booksForm/booksForm.js]: ## Move the book component
- [x] The `bookForm.js` component needs to be moved to the `container` directory. You only need to have the `App.js` and the `book.js` components in this directory.

File: src/components/booksForm/booksForm.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+const BookForm = ({ addBook }) => {
+  const handleForm = (event) => {
+    const title = event.target.title.value;
+    const categorie = event.target.categorie.value;
+    addBook({
+      title,
+      categorie,
+    });
+    event.preventDefault();
+  };
```
Comment (christianotieno) [inline src/components/booksForm/booksForm.js]: ## Implement the missing book id section 
- [x] Our `book` component should represent a single row of books with `id`, `title` and `category` [screenshot from Pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone2/Book/RepresentARowInBooksTable.png). We are currently missing the `id` part, let's implement it here. We can have this by generating a random number to act as our `id` whenever a book gets created. that's one way to do it.

File: src/components/booksList/booksList.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
+/* eslint-disable arrow-body-style, max-len, react/forbid-prop-types */
```
Comment (christianotieno) [inline src/components/booksList/booksList.js]: ## Move the bookList component
- [x] The `bookList.js` component needs to be moved to the `container` directory. You only need to have the `App.js` and the `book.js` components in this directory.

File: src/containers/actions/index.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
+/* eslint-disable arrow-body-style */
+
```
Comment (christianotieno) [inline src/containers/actions/index.js]: ## Move the action files
- [x] The `action.js` module needs to be moved to the `src` directory. You only need to have the `bookForm.js` and the `bookList.js` components in this directory.

File: src/containers/reducers/book.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
+/* eslint-disable no-return-assign */
```
Comment (christianotieno) [inline src/containers/reducers/book.js]: ## Move the reducer directory
- [x] The `reducer` directory needs to be moved to the `src` directory. You only need to have the `bookForm.js` and the `bookList.js` components in this directory.

**Review by christianotieno — CHANGES_REQUESTED** [review]
## Changes Required ♻️ 
Hello @pastorp3 👋🏾 
Congratulations on submitting the second milestone for this project for review. Take note of the following required changes before we proceed. For this milestone, we need to make sure the correct order of where we place our code is followed.

### Mandatory
- [x] Move the `App.js` to the `components` directory.
- [x] Add the missing book `id` in the `book` component.
- [x] Move the `bookList.js` and the `bookForm.js` components to the container directory.
- [ ] Move the `actions` and `reducers` directories to the src directory.

### Happy coding 💻 
#### Christian Otieno (CR)

**Review by uche-inyama — CHANGES_REQUESTED** [review]
## Changes Required

### Hi @pastorp3 :wave: 

* [ ] What is left from the changes requested from the previous reviewer is the **id**, in the book object. The book object that is created in the bookform component should look like so:


```
  addBook({
    id: generateID(),
    title,
    category
  });
```
You can use the [uuid package](https://www.npmjs.com/package/uuid) to generate the random ids.

#### Make this change, then submit another form for a code review


### Happy Coding :technologist: 

#### ``` Uchechukwu Henry inyama ```

**Review by wathigo — APPROVED** [review]
## APPROVED :heavy_check_mark:  
Great job implementing the suggested changes.
All requirements have been met.

**You may proceed to the next milestone.**

**Happy Coding :computer: 
Simon Wathigo**

**Pull request merged**